tests/ui_tools/test_popups.py

```diff
@@ -343,14 +355,66 @@ def test__fetch_user_data(
 
         mocker.patch.object(self.controller.model, "get_user_info", return_value=data)
 
-        display_data = self.user_info_view._fetch_user_data(self.controller, 1)
+        display_data, custom_profile_data = self.user_info_view._fetch_user_data(
+            self.controller, 1
+        )
 
         assert display_data.get(expected_key, None) == expected_value
 
+    @pytest.mark.parametrize(
+        [
+            "to_vary_in_each_user",
+            "expected_value",
+        ],
+        [
+            case(
+                [],
+                {},
+                id="user_has_no_custom_profile_data",
+            ),
+            case(
+                [
+                    {
+                        "label": "Biography",
+                        "value": "Simplicity",
+                        "type": 2,
+                        "order": 2,
+                    },
+                    {
+                        "label": "Mentor",
+                        "value": [11, 12],
+                        "type": 6,
+                        "order": 7,
+                    },
+                ],
+                {"Biography": "Simplicity", "Mentor": "Human 1, Human 2"},
+                id="user_has_custom_profile_data",
+            ),
+        ],
+    )
+    def test__fetch_user_data_custom_profile_data(
+        self,
+        mocker: MockerFixture,
+        to_vary_in_each_user: Dict[str, Any],
```

Comment:
This is the wrong type. Unfortunately mypy/pytest doesn't check the parametrize(s) to the parameters.